### ISSUE TITLE
Error 401

### ISSUE THREAD

**Issue by TomsnCo** [PfisterDaniel/node-red-contrib-apple-find-me#30]
Hello,
Since today I have a 401 error on my flows, 
The only log is: { status: false, code: 401, data: null },

<img width="294" alt="Capture d’écran 2022-10-05 à 20 53 50" src="https://user-images.githubusercontent.com/56442712/194139553-c96f15e9-5e06-45ed-82b0-65ce7f71177e.png">

**Comment by PfisterDaniel** [comment]
Yes i have the same problem. The error comes from the Apple Service, i can't fix it at the moment... i hope tomorrow works again.

**Comment by acaranta** [comment]
mmmh tried re-entering password and/or changing to "app specific password" ... that didn't help 🙈 could they possibly have locked us out of their API ? 😩

**Comment by dhanzseros** [comment]
I hope they haven't locked. This is the only one node to get datas from my phone....

**Comment by brinkgit** [comment]
Same problem since 5-10 18:03 (Amsterdam time). Update to version 1.1.8 did not help (was stil on 1.1.5 or 6)

**Comment by hijamie32** [comment]
I am getting a similar issue but Code 404 on two separate accounts. Are they linked otherwise will open a new issue?

Thanks

**Comment by OmegaQuest** [comment]
Getting same error, 401 on all Apple Locate Input Nodes since October 5th @ around 11:15 am CST.  I also tried upgrading to 1.1.8, recreating the Nodes from Scratch, and none of that solve the issue.

**Comment by brinkgit** [comment]
Seems that error 401 has something tot do with old certificates? But I don't know how to install new certificates or how to make this app loading a new certificate.

**Comment by PfisterDaniel** [comment]
No, certificates have nothing to do with it. Currently, the node is not working. Probably due to a change from Apple.

**Comment by Michl7** [comment]
The openHAB icloud binding also stopped working. https://github.com/openhab/openhab-addons/issues/13282
Looks like a change or problem on apples side.

**Comment by rmds2000** [comment]
Did anyone find a solution for the node-red-contrib-apple-find-me error message 401 - Not authorised

**Comment by Supergiovane** [comment]
I've searched but it seems to be a common issue. I'm fearing, we will not be able to access such findme APIs anymore.
Fortunately, i've Unifi stuffs and i can do a presence detection using Unifi node.
As alternative, you can set an automation directly on your phone, that turns on or off a switch, whenever you enter/leave your home, that you can then query for the status using node-red.

**Comment by rmds2000** [comment]
Really sad it was a good node

**Comment by hijamie32** [comment]
I don't know about the Find Me API's as the ICloud3 integration is still working well?

**Comment by Supergiovane** [comment]
> I don't know about the Find Me API's as the ICloud3 integration is still working well?

Maybe due to the 2FA authentication? It can be.

**Comment by brinkgit** [comment]
Find-my-iPhone does (as far as I know) not need 2FA. On the iCloud site you can start find-my-phone without the 2FA code.

**Comment by edautz** [comment]
The Homeassistant ICloud and ICloud3 integrations are working well, also after this problem has occurred. As far I know these are the only solutions that remain.

**Comment by na-am** [comment]
i modified my flows to "owntracks" so all my dashboards and logics are working again (Node-red, influxdb, grafana)
https://apps.apple.com/de/app/owntracks/id692424691

pros:
- radius per region
- much more infos, see https://owntracks.org/booklet/tech/json/

cons:
- must be installed on every device
- mqtt broker needed
- eventually battery consumption

**Comment by Supergiovane** [comment]
> The Homeassistant ICloud and ICloud3 integrations are working well, also after this problem has occurred. As far I know these are the only solutions that remain.

I think they're using 2FA or, most probably, 2SA auth. They're using icloud.com apis as well.
I've not seen the code yet, but the simplest way is as @edautz wrote... install Homeassistant and use this. In the meantime, i hope PfisterDaniel will fix the issue, but i know... working with such APIs is far from being simple.

**Comment by MikeJ67** [comment]
Has anyone another way to read the battery status of an ipad from within Node-Red? I don't need location, only the battery-value of that one device (an info panel on the wall, which turns its own loader power on and off. And no, shortcuts is no option)

**Comment by edautz** [comment]
> Has anyone another way to read the battery status of an ipad from within Node-Red? I don't need location, only the battery-value of that one device (an info panel on the wall, which turns its own loader power on and off. And no, shortcuts is no option)

If you install Homeassistant and install ICloud (the easiest) or ICloud3 integration and install within Nodered node-red-contrib-home-assistant-websocket you can readout the Homeassistant sensors including the iCloud device battery levels, also gps positions etc.   in Nodered.

**Comment by OmegaQuest** [comment]
Hope this issue gets resolved/work around soon.  I dont use Homeassistant and only Node-Red and MQTT but need the GPS (Coordinates) for Family Tracking.  I've tried OwnTracks but not a fan of it due to Battery Drain and an additional app that needs to be installed and maintained.

**Comment by brinkgit** [comment]
Anyone any kind of solution?

**Comment by Michl7** [comment]
Looks like they already work on the icloud binding for openhab ( https://github.com/openhab/openhab-addons/issues/13282 )which also not working anymore. So i think a rework of the node needs to happen.

**Comment by xiaonan902198** [comment]
401认证报错这个问题有办法解决么？

**Comment by hazymat** [comment]
Same issue - error 401, last worked 22 days ago.

**Comment by wizy69** [comment]
same issue here, still exists 03112022. Awesome Node for Find Me Message Broadcast

**Comment by brinkgit** [comment]
Something has changed the last few days. It was '401 not authorised', now it says 'ErrorCode: -1'.

**Comment by demokedes** [comment]
Just checked right now and it is still "401 - Not authorised".

**Comment by brinkgit** [comment]
Right, it turned back on the 401 message on my system too.

**Comment by zyce78** [comment]
@PfisterDaniel, seem to be some progress over at https://github.com/openhab/openhab-addons/issues/13282. If it could be of any help. Thanks for your work so far of this plugin!

**Comment by OmegaQuest** [comment]
@PfisterDaniel, Any updates on this issue?

**Comment by brinkgit** [comment]
Does anyone have any sort of solution for this? My programming skills are not good enough to solve this issue. Anyone else?

**Comment by na-am** [comment]
> Does anyone have any sort of solution for this? My programming skills are not good enough to solve this issue. Anyone else?

as i mentioned two month ago "owntrack" could be a alternative
or
if you have an apple computer running 7/24 with Node-red (FindMe App has to be started as well) you can easily grab your devices and airtags locations with the attached Node-red flows
[airtags_devices.json.zip](https://github.com/PfisterDaniel/node-red-contrib-apple-find-me/files/10245615/airtags_devices.json.zip)

**Comment by Kimovitzh** [comment]
There seems to be hope!

https://github.com/openhab/openhab-addons/issues/13282

**Comment by theAnDrO40** [comment]
If there is some interest i've got a little workaround with mqtt python icloud-home-4

**Comment by OmegaQuest** [comment]
> icloud-home-4

I am, although I do not use Home Assistant and only use Node-Red and MQTT.

**Comment by theAnDrO40** [comment]
This is a python script and this can collect data from the icloud devices and with node red you can send these information into the mqtt protocol

**Comment by OmegaQuest** [comment]
> This is a python script and this can collect data from the icloud devices and with node red you can send these information into the mqtt protocol

Cool, where might I find this:  "icloud-home-4"  script.  I looked on Google with no luck.

**Comment by regissg** [comment]
The Openhab guys seem to have found a fix to rework authentication to reflect changes in iCloud API https://github.com/openhab/openhab-addons/pull/13691 any hope on the node red side?

**Comment by OmegaQuest** [comment]
I really wish this would get fixed/updated. OwnTacks is not accurate enough due to Apples 5 Min max motion updates when OwnTracks is only running in the background.

**Comment by regissg** [comment]
An easy workaround for me was to install openhab just to run the icloud binding and to add the node-red-contrib-openhab3 nodes to use the "find my phone" feature

**Comment by jjhall99** [comment]
even with 2FA and an app specific password for node-red created, this still fails. hoping not to need to go through all of the other hoops...

**Comment by davidcreager** [comment]
Ok, so from my perspective the best/cleanest working javascript icloud api repo is https://github.com/foxt/icloud.js

Couple of things to note -:

There is a bug that needs to be fixed (although it is very easy to work around) whereby, refreshing devices, needs a re-authentication. (I have hacked a solution which works and is only a few lines of code)

the approach taken to authenticating with. apple requires the entry of a 2FA if the session is untrusted (it persists the session, so it is only required occasionally) I don't really understand how apple-find me did not need this.

I plan to put a simple node-red wrapper around this (will be a long way from the apple-findme sophistication, but it will support my use case.) For the code, I am considering two approaches, either injecting the code into the node or having a standalone script with input that can be run

I would much prefer to use apple-findme, and will keep following to see if it gets fixed

Good luck and thanks sharing your work, I very much appreciate the use I got from apple-findme

If I can help with apple-find me, then please let me know

**Comment by TomsnCo** [comment]
As of now, I'm still stuck on locating my phone from nodeRED,
Would you have a support to circumvent the problem encountered?

With pleasure,
THANKS,
